### ISSUE TITLE
Crash when searching

### ISSUE THREAD

**Issue by i7u** [WhisperSystems/Signal-Android#6048]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
Signal crashes when I use the search function.

### Steps to reproduce
- Open the app
- Tap the magnifying glass icon on the top of the screen to search
- The "Search..." bar comes up
- Type something into the search bar
- After one or more searches, while the app is sorting the results, the app will crash.

**Actual result:** The app crashes.
**Expected result:** Search results are returned in real time as the search string is changed.

### Device info
**Device:** Samsung Galaxy S5 Neo
**Android version:** 6.0.1
**Signal version:** 3.26.2

### Link to debug log
Crash 1 (immediate as soon as I started typing in search bar): https://gist.github.com/64bfebc10d6b1f5c6690fb67de21101d

Crash 2 (as first search results were being returned): https://gist.github.com/c1517ddb4688225c496f4b2c2251d531

Crash 3 (after multiple search strings were inputed/deleted in search bar):  https://gist.github.com/b2e4b5e1f66c3ced2690ea4d0972769e

**Comment by 2-4601** [comment]
```
E AndroidRuntime: FATAL EXCEPTION: ModernAsyncTask #1
E AndroidRuntime: Process: org.thoughtcrime.securesms, PID: 13984
E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
E AndroidRuntime: 	at android.support.v4.content.ModernAsyncTask$3.done(ModernAsyncTask.java:161)
E AndroidRuntime: 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
E AndroidRuntime: 	at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:818)
E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to write to field 'java.util.LinkedHashMap$LinkedEntry java.util.LinkedHashMap$LinkedEntry.nxt' on a null object reference
E AndroidRuntime: 	at java.util.LinkedHashMap.addNewEntry(LinkedHashMap.java:197)
E AndroidRuntime: 	at java.util.HashMap.put(HashMap.java:404)
E AndroidRuntime: 	at org.thoughtcrime.securesms.database.CanonicalAddressDatabase.getCanonicalAddressId(CanonicalAddressDatabase.java:168)
E AndroidRuntime: 	at org.thoughtcrime.securesms.database.CanonicalAddressDatabase.getCanonicalAddressIds(CanonicalAddressDatabase.java:188)
E AndroidRuntime: 	at org.thoughtcrime.securesms.database.ThreadDatabase.getFilteredConversationList(ThreadDatabase.java:305)
E AndroidRuntime: 	at org.thoughtcrime.securesms.database.loaders.ConversationListLoader.getFilteredConversationList(ConversationListLoader.java:65)
E AndroidRuntime: 	at org.thoughtcrime.securesms.database.loaders.ConversationListLoader.getCursor(ConversationListLoader.java:29)
E AndroidRuntime: 	at org.thoughtcrime.securesms.util.AbstractCursorLoader.loadInBackground(AbstractCursorLoader.java:70)
E AndroidRuntime: 	at org.thoughtcrime.securesms.util.AbstractCursorLoader.loadInBackground(AbstractCursorLoader.java:11)
E AndroidRuntime: 	at android.support.v4.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:299)
E AndroidRuntime: 	at android.support.v4.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:57)
E AndroidRuntime: 	at android.support.v4.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:45)
E AndroidRuntime: 	at android.support.v4.content.ModernAsyncTask$2.call(ModernAsyncTask.java:138)
E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E AndroidRuntime: 	... 3 more
```

**Comment by 2-4601** [comment]
Oh, I think this is already fixed edac0e85c76edb6c6632481611794cf13bb8456c
Closing as duplicate of #6044
